server.js

```diff
@@ -1,34 +1,107 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
+import listEndpoints from 'express-list-endpoints'
+
+import booksData from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+app.get('/books', (req, res) => {
+  const { author, title, lang } = req.query
+  let booksResult = booksData
+
+  if (author) {
+    booksResult = booksResult
+      .filter((book) => book.authors.toLowerCase().includes(author.toLowerCase()))
+  } 
+  
+  if (title) {
+    booksResult = booksResult
+      .filter((book) => book.title.toLowerCase().includes(title.toLowerCase()))
+  } 
+  
+  if (lang) {
+    booksResult = booksResult.filter((book) => book.language_code === lang)
+  }
+  res.json(booksResult)
+})
+
+app.get('/books/id/:id', (req, res) => {
```

Comment:
when you use a primary key, like id to filter your data it is the convention to not put key name in the address so it should be `/books/:id`